src/service-api.ts

```diff
@@ -19,24 +20,86 @@ const plugin: FastifyPluginAsync = async (fastify) => {
   // get current user
   fastify.get('/current', async ({ member }) => member);
 
-    // // get member
-    // fastify.get(
-    //   '/:id',
-    //   { schema: getOne },
-    //   async ({ category, id , log }) => {
-    //     const task = taskManager.createGetTask(category, id);
-    //     return runner.runSingle(task, log);
-    //   },
-    // );
-
-    // get members
+    // get category name (age group)
+    fastify.get<{ Params: {categoryId: string };}>(
+      '/category/age/:categoryId',
+      async ({ member, params: {categoryId}, log }) => {
+        const task = taskManager.createGetTask(member, categoryId);
+        return runner.runSingle(task, log);
+      },
+    );
+
+    //get category (discipline)
+    fastify.get<{ Params: {categoryId: string };}>(
+      '/category/discipline/:categoryId',
+      async ({ member, params: {categoryId}, log }) => {
+        const task = taskManager.createGetCategoryDiscTask(member, categoryId);
+        return runner.runSingle(task, log);
+      },
+    );
+
+    // get age categories
     fastify.get(
-      '/allcategories',
+      '/categories/age',
       async ({ member, log }) => {
         const task = taskManager.createGetAllTask(member);
         return runner.runSingle(task, log);
       },
     );
+
+        // get discipline categories
+    fastify.get(
+      '/categories/discipline',
+      async ({ member, log }) => {
+        const task = taskManager.createGetAllDisciplinesTask(member);
+        return runner.runSingle(task, log);
+      },
+    );
+
+    // get items in one category (age)
+    fastify.get<{ Params: {categoryId: string };}>(
+      '/age/:categoryId',
+      async ({ member, params: {categoryId}, log }) => {
+        const task = taskManager.createGetItemsByAge(member, categoryId);
+        return runner.runSingle(task, log);
+      },
+    );
+
+    //get category (discipline)
+    fastify.get<{ Params: {categoryId: string };}>(
+      '/discipline/:categoryId',
+      async ({ member, params: {categoryId}, log }) => {
+        const task = taskManager.createGetItemsByDisc(member, categoryId);
+        return runner.runSingle(task, log);
+      },
+    );
+
+    //get category of an item
+    fastify.get<{ Params: {itemId: string };}>(
+      '/:itemId/category',
```

Comment:
With my suggestion, this will become `/:itemId/categories` and will return an array.